token/program-2022/src/instruction.rs

```diff
@@ -569,6 +569,12 @@ pub enum TokenInstruction {
         /// Maximum fee assessed on transfers
         maximum_fee: u64,
     },
+
+    /// The common instruction prefix for Confidential Transfer extension instructions.
+    ///
+    /// See `extension::confidential_transfer::instruction::ConfidentialTransferInstruction` for
+    /// further details about the extended instructions that share this instruction prefix
+    ConfidentialTransferExtension,
```

Comment:
I initially thought we would directly add all of the confidential instructions here, but your approach neatly encapsulates them.  It makes discoverability of instructions trickier, but not very much.

If we do it here, however, perhaps we should have all of the extension instructions use this same model, so all of the `TransferFee`-related instructions are in their own namespace, and so on.  Once people start forking SPL token, separate namespaces for extensions will make their work much nicer, since they can just quickly throw an `unimplemented!()` for a whole set of instructions if they choose.  What do you think?

Reply:
> I initially thought we would directly add all of the confidential instructions here, but your approach neatly encapsulates them.

My main motivation was to avoid getting tangled in the manual unpacking code of the top-level instructions.  I had nightmares about having to juggle the ordinals as instructions from multiple extensions merged in at the same time!

Yep for sure, the TransferFee extension instructions probably could be implemented like this as well

Reply:
> perhaps we should have all of the extension instructions use this same model

I like that idea